thewalrus/tests/test_internal_modes.py

```diff
@@ -1171,7 +1172,8 @@ def test_mixed_heralded_photon(nh):
     assert np.allclose(np.diag(dm_modeb), p_b)
 
 
-def test_pure_gkp():
+@pytest.mark.parametrize("method", ["recursive", "non-recursive"])
```

Comment:
This test fails for the recursive method, which is exactly why we added the non-recursive. So I suggest removing the parametrize here and just run it for the recursive that works. Then we should also be able to lower the tolerances.